sycl/include/sycl/handler.hpp

```diff
@@ -381,6 +381,15 @@ class __SYCL_EXPORT handler {
                             "a single kernel or explicit memory operation.");
   }
 
+  enum SyclExtensions {
+    sycl_ext_oneapi_kernel_properties,
+    sycl_ext_oneapi_enqueue_barrier,
+    sycl_ext_oneapi_memcpy2d,
+    sycl_ext_oneapi_device_global
+  };
+
+  template <SyclExtensions ExtensionT> void throwIfGraphAssociated();
```

Comment:
This is an ABI change, but we haven't updated the `abi/sycl_symbols_linux.dump`. Also need to check whether this would affect the ability of this PR to be cherry-picked as a bug fix

```
There are new symbols in the new library. It is a non-breaking change. Refer to sycl/doc/developer/ABIPolicyGuide.md for further instructions.
The following symbols are new to the object file:

_ZN4sycl3_V17handler22throwIfGraphAssociatedILNS1_14SyclExtensionsE0EEEvv
_ZN4sycl3_V17handler22throwIfGraphAssociatedILNS1_14SyclExtensionsE3EEEvv
_ZN4sycl3_V17handler22throwIfGraphAssociatedILNS1_14SyclExtensionsE2EEEvv
_ZN4sycl3_V17handler22throwIfGraphAssociatedILNS1_14SyclExtensionsE1EEEvv
```

Reply:
I updated this abi/sycl_symbols_linux.dump with the new functions

Reply:
Nice, I tried this on git branch on Windows and its symbol test passed without any further changes